server/dataone_upload.py

```diff
@@ -167,26 +222,26 @@ def create_upload_package(item_ids, tale, user, repository):
     """
 
     logger.debug('Entered create_upload_package')
-    filtered_items = filter_input_items(item_ids, user)
-    dataone_objects = filtered_items['dataone']
+    filtered_items = filter_items(item_ids, user)
+    dataone_object_pids = filtered_items['dataone']
     local_objects = filtered_items['local']
+    # globus_objects = filtered_items['globus']
     local_file_pids = list()
 
     try:
-        print(type(repository))
         client = create_client(repository, {"headers": {
-            "Authorization": "Bearer <TOKEN>"}})
-
-        # for object in dataone_urls:
-        #    dataone_pids.append(create_upload_object_metadata(client, object))
+            "Authorization": "Bearer eyJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJodHRwOlwvXC9vcmNpZC5vcmdcLzAwMDAtMDAwMi0xNzU2LTIxMjgiLCJmdWxsTmFtZSI6IlRob21hcyBUaGVsZW4iLCJpc3N1ZWRBdCI6IjIwMTgtMDUtMTlUMTc6MTM6MTguMzgzKzAwOjAwIiwiY29uc3VtZXJLZXkiOiJ0aGVjb25zdW1lcmtleSIsImV4cCI6MTUyNjgxNDc5OCwidXNlcklkIjoiaHR0cDpcL1wvb3JjaWQub3JnXC8wMDAwLTAwMDItMTc1Ni0yMTI4IiwidHRsIjo2NDgwMCwiaWF0IjoxNTI2NzQ5OTk4fQ.g5lWjRqG71V0jaeOR8_v3KRWwkITQuVLt76_DYt6RvR9xVkLLdOcsB654SmbFZykrF7Hr4OityDXsu1U-eTRUvTTwsSlG3KyO1llmudcowFQ_CiYsTF0RJxIywwLhPbcXBWy8tjTgxA0Ni7rVNP4v_N8y9UQ1C0BvNVFp9RktLYik0u2tGYrdiotLIsVlNhvUwwjvrdHJ3QHlQIoi34TttIyss4O04iwVN9_wasS3YDrBQRWKkH3hJRcWQXHjcmd1-LUo7ek-cIL_ckcnmg4yV7Ods6FOrfK1Pz56IS-MWYAs1sJq5vWB2OxW1YGNryUwE5D7tf_bLdzO65BL-ZYfEvMUSnggliN6AxpX2dfGGEtBXAvcggX70jm0gR11zK46-bSQLf8ZLin0ZhtZTiPZ8X86NpJVZqmUUpEGU0dQGr5eSq9CrobdVbvyEPd4srwdiqfwVrafjJc-JevE0314VFhQjeko26UezPofUcXUmVT9x6Ydp-RiNdiAPImj3tnZTmPdZ-F8QFSoXj35PjCZbRUFk4SbFsl5OMHNfOo4pVNqSxJR14BW5OEA8UOVzYpdjes0-LsazkQmSsuCYj1F7Bn3lylxE9UqK4Jy1aNRFBs_DMrli2SCiqGf1iyfcme1R1hdSu0T8YZTkz4X6tcsN2fgepsJk6_rwGxg3kwu_o"}})
 
         for file in local_objects:
             local_file_pids.append(create_upload_object_metadata(client, file))
 
-        eml_pid = create_upload_eml(tale, client, user)
-        # Add local_file_pids to dataone_objects when it is time to support local files.
-        create_upload_resmap(str(uuid.uuid4()), eml_pid, dataone_objects+local_file_pids, client)
-    except DataONEException as e:
-        logger.debug('DataONE Error: {0}'.format(e))
+        eml_pid = create_upload_eml(tale, client, user, item_ids)
 
+        # Combine the lists of pids into a single list that will be used in the resource map
+        upload_objects = dataone_object_pids+local_file_pids
+        create_upload_resmap(str(uuid.uuid4()), eml_pid, upload_objects, client)
+
+    except DataONEException as e:
+        logger.debug('DataONE Error: {}'.format(e))
+        raise ValidationException('Error uploading file to DataONE. {0}'.format(str(e)))
```

Comment:
Is raising a `girder.models.model_base.ValidationException` here the Girder-ish way to do things to signal an API failure? Or is there another Girder or d1_python exception you could raise?